### ISSUE TITLE
Stratified internal validation set for xgboost

### ISSUE THREAD

**Issue by fdeoliveirag** [tidymodels/parsnip#760]
While working with boosted trees, wouldn't it be interesting to allow the validation set to follow the same stratification defined in the splits/folds?

Currently it's only possible to determine the _proportion_ of the data that will be used in the validation:

```r
library(parsnip)

# e.g. 20% of training set
valid_prop <- 0.2

xgb_spec <- boost_tree() %>%
  set_engine("xgboost", validation = valid_prop) %>%
  set_mode("classification")
```

The _slicing_ itself will be necessarily random:

https://github.com/tidymodels/parsnip/blob/b4f065b6cb319885cd9328888e5ba3d2318cabb8/R/boost_tree.R#L402-L412

I think this could be relevant specially for highly imbalanced class data.
